src/overlays/actors/ovl_En_Dnh/z_en_dnh.c

```diff
@@ -31,28 +52,129 @@ ActorInit En_Dnh_InitVars = {
     (ActorFunc)EnDnh_Draw,
 };
 
-#endif
-
-extern UNK_TYPE D_06002950;
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dnh/func_80A50D40.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dnh/func_80A50DF8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dnh/func_80A50E40.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dnh/func_80A50EC0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dnh/func_80A50F38.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dnh/func_80A50F9C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dnh/EnDnh_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dnh/EnDnh_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dnh/EnDnh_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dnh/func_80A51168.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dnh/EnDnh_Draw.s")
+static AnimationInfoS sAnimationInfo[] = { &object_tro_Anim_0000A0, 1.0f, 0, -1, ANIMMODE_LOOP, 0 };
+
+UNK_TYPE D_80A5143C[] = { 0x06001140, 0x06001940, 0x06002140, 0x06001940, 0x00000000 };
+
+void* func_80A50D40(Actor* actor, PlayState* play) {
+    func_800B7298(play, actor, 7U);
+    if (gSaveContext.eventInf[3] & 0x20) {
+        play->nextEntrance = 0xC60;
+    } else {
+        play->nextEntrance = 0x8460;
+    }
+    gSaveContext.nextCutsceneIndex = 0;
+    play->transitionTrigger = 0x14;
+    play->transitionType = 3;
+    gSaveContext.nextTransitionType = 7;
+    gSaveContext.save.saveInfo.weekEventReg[0x5A] &= 0xBF;
+    gSaveContext.eventInf[5] &= 0xFE;
+    return 1;
+}
+
+void* func_80A50DF8(EnDnh* this, PlayState* play) {
+    switch (this->unk198) {
+        case 1:
+            return &D_80A51384;
+        case 2:
+            return &D_80A5138C;
+        default:
+            return &D_80A51250;
+    }
+}
+
+s32 func_80A50E40(EnDnh* this, PlayState* play) {
+    if (!(this->unk18C & 7) || (Actor_ProcessTalkRequest(&this->actor, &play->state) == 0)) {
+        return 0;
+    }
+    SubS_UpdateFlags(&this->unk18C, 0U, 7U);
+    this->unk190 = func_80A50DF8(this, play);
+    this->actionFunc = func_80A50F38;
+    return 1;
+}
+
+s32 func_80A50EC0(EnDnh* this) {
+    s16 var_v1;
+
+    if (this->unk19C == 0) {
+        var_v1 = 0;
+    } else {
+        this->unk19C = this->unk19C - 1;
+        var_v1 = this->unk19C;
+    }
+    if (var_v1 == 0) {
+        this->unk19E += 1;
+        if (this->unk19E >= 4) {
+            this->unk19C = Rand_S16Offset(0x1E, 0x1E);
+            this->unk19E = 0;
+        }
+    }
+}
+
+void func_80A50F38(EnDnh* this, PlayState* play) {
+    if (func_8010BF58(&this->actor, play, this->unk190, this->unk1A0, &this->unk194) != 0) {
```

Comment:
```suggestion
    if (func_8010BF58(&this->actor, play, this->msgEventScript, this->unk1A0, &this->unk194)) {
```
This is a function related to the message event script which has not been documented or merged into master yet, but this is the name we used. We aren't sure on the type yet, but `msgEventScript` should at least be a pointer. `s32*` should be fine to type this